python/S3_BUCKET_NAMING_CONVENTION/S3_BUCKET_NAMING_CONVENTION.py

```diff
@@ -0,0 +1,417 @@
+# Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+
+Rule Name:
+  S3_BUCKET_NAMING_CONVENTION
+
+Description:
+  Verify that S3 Bucket name conforms to a particular regex pattern.
+
+Trigger:
+  Configuration change
+
+Reports on:
+  AWS::S3::Bucket
+
+Rule Parameters:
+  regexPattern (Required)
+    The string value provided to check with the S3 bucket name.
+
+Scenarios:
+  Scenario 1:
+  Given: No S3 buckets present
+   Then: Return NOT_APPLICABLE
+
+  Scenario 2:
+  Given: Rule parameter regexPattern is not configured or not valid
+   Then: Return ERROR
+
+  Scenario 3:
+  Given: At least 1 S3 bucket present
+    And: Rule parameter regexPattern is configured and valid
+    And: The Bucket name match the regexPattern
+   Then: Return COMPLIANT
+
+  Scenario 4:
+  Given: At least 1 S3 bucket present
+    And: Rule parameter regexPattern is configured and valid
+    And: The Bucket name does not match the regexPattern
+   Then: Return NON_COMPLIANT
+
+'''
+
+import json
+import sys
+import datetime
+import re
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+
+    s3_client = get_client('s3', event)
+    list_of_buckets = s3_client.list_buckets()
```

Comment:
As this is a configuration change triggered rule, we don't need to check whether at least one bucket exist or not. Config will handle this.